tools/c7n_gcp/c7n_gcp/resources/sql.py

```diff
@@ -118,6 +118,24 @@ def get_resource_params(self, model, resource):
                 'body': {'settings': {'activationPolicy': 'ALWAYS'}}}
 
 
+@SqlInstance.action_registry.register('enable-deletion')
+class SqlInstanceEnableDeletion(MethodAction):
+
+    schema = type_schema(
+        'enable-deletion',
+        value={'type': 'string'})
+    method_spec = {'op': 'patch'}
+    path_param_re = re.compile('.*?/projects/(.*?)/instances/(.*)')
+    method_perm = 'enable-deletion'
+
+    def get_resource_params(self, model, resource):
+        project, instance = self.path_param_re.match(
+            resource['selfLink']).groups()
+        return {'project': project,
+                'instance': instance,
+                'body': {'settings': {'deletionProtectionEnabled': self.data['value']}}}
```

Comment:
hmmm.. we have historically just done this as a force flag on the delete action.

Reply:
If deletion protection is enabled for gcp sql instance, it doesn't allow us to delete the resources. For that purpose, I added this functionality to disable deletion protection first and then delete the resource.

Reply:
the result is the same, the question is how to expose this to the user, if their intent is to delete irrespective of deletion protection, simply setting a flag on the delete action is the clearest way to express the intent versus having to use a separate action. I've gone ahead and pushed the changes to enable that in this pr.

Reply:
actually it does feel like there might be some utility to have the action separate for the use of wanting to turn on deletion protection.

Reply:
added back the action renamed to set-deletion-protection with a boolean value instead of string, and left the force flag on delete.